tests/test_dojo.py

```diff
@@ -76,21 +116,146 @@ def test_print_room(self):
+        res = self.dojo.reallocate_person('kimani', 'swift')
+        self.assertEqual(res,None)        
+
+    def test_reallocate_to_same_room(self):
+        self.dojo.create_room('hogwarts','office')
+        self.dojo.create_room('swift', 'livingspace')
+        self.dojo.add_person("john", "fellow")
+        res = self.dojo.reallocate_person('joe', 'hogwarts')
+        self.assertEqual(res,None)  
+
+    def test_a_person_has_been_removed_from_a_room_after_reallocation(self):
+        self.dojo.create_room('hogwarts', 'office')
+        self.dojo.add_person('wick', 'fellow')
+        office = self.dojo.dojo_office[0]
+        members = office.members
+        self.assertEqual(len(members), 1)
+        self.dojo.create_room('php', 'office')
+        self.dojo.reallocate_person('wick', 'php')
+        office = self.dojo.dojo_office[0]
+        members = office.members
+        self.assertEqual(len(members), 0)
+
+    def test_if_a_room_doesnt_exist(self):
+        self.dojo.create_room('hogwarts','office')
+        self.dojo.create_room('swift', 'livingspace')
+        self.dojo.add_person("john", "fellow")
+        res = self.dojo.reallocate_person('joe', 'hogwarts3')
+        self.assertEqual(res,None) 
+
+        # test for load people from file --->
+    def test_load_people(self):
+        self.dojo.create_room('hogwarts', 'office')
+        self.dojo.create_room('swift', 'livingspace')
+        staff = self.dojo.staff
+        fellows = self.dojo.fellows
+        self.assertEqual(len(staff), 0)
+        self.assertEqual(len(fellows), 0)
+        self.dojo.load_people("sample")
+        self.assertEqual(len(staff), 3)
+        self.assertEqual(len(fellows), 4)
+
+        # tests for load and save data -->
+    def load_data(self):
+        self.dojo.create_room('hogwarts', 'office')
+        self.dojo.create_room('swift', 'livingspace')
+        self.dojo.add_person('burney', 'fellow', 'Y')
+        self.dojo.add_person('martial', 'staff')
+        return
+
+    def test_saves_state(self):
+        data = self.load_data()
+        create_db("databasefiles/test.db")
+        self.dojo.save_state('test')
+        # check if file is created
+        self.assertFalse(os.path.exists("databasefiles/test.db"))
+
+    def test_load_state(self):
+        '''
+        Testss that data is actually loaded
+        from an existing database
+        '''
+        #populate database
+        data = self.load_data()
+        # save state to file
+        self.dojo.save_state('test_load.db')
+        # loads data from the saved database
+        self.dojo.load_state('test_load.db')
+        # Entry of data to the system
+        self.assertEqual(2, len(self.dojo.all_people))
+        self.assertEqual(1, len(self.dojo.fellows))
+        self.assertEqual(1, len(self.dojo.staff))
+        self.assertEqual(2, len(self.dojo.all_rooms))
+        self.assertEqual(1, len(self.dojo.livingspace))
+        self.assertEqual(1, len(self.dojo.office))
```

Comment:
assert statements should generally be, (output, expected output) and not (expected output, output)